### DISCUSSION TITLE
Allow field type "Dropdown (Multiple)" to require a selection

### DISCUSSION THREAD

**Discussion by JasonAGross** [directus/directus#14640] [Feature Requests]
Multi-select dropdown fields currently have no way of enforcing that at least one option is selected. The interface option of "Allow None" will provide the option to deselect all choices if enabled. However, if disabled the user can still deselect all dropdown choices without generating an error or warning. By not selecting "Allow No Selection" in the interface setup, it implies that a selection must be made for this field. 

Furthermore, validation filters for this field type are limited to "is null" or "isn't null" which has no function since a fully deselected dropdown will result in an empty array, preventing a "null" condition from ever being met. 

At a minimum the term "Allow No Selection" should be reworked to reflect what that option actually does, which seems to only provide an interface through which the user can quickly unselect all options. What would be better would be to show an error on the collection screen, preventing a collection from being created/updated, when no choices are selected on a dropdown that has "Allow No Selection" unchecked. 

An alternative could be to provide "is empty" and "is not empty" filter options for the validation rules associated with this field type to better reflect it's array-driven value. 

### To Reproduce

Create a new collection with a "Dropdown (Multiple)" field and include at least one choice for that field. Ensure "Allow No Selection" is unchecked in the field interface setup. In the collection, deselection of all choices is still allowed. 

### Errors Shown

None

### What version of Directus are you using?

9.14.2

### What version of Node.js are you using?

16.13.1

### What database are you using?

Postgres 14

### What browser are you using?

Chrome

### How are you deploying Directus?

Heroku

**Comment by rijkvanzanten** [comment]
Have you tried setting the field to "allow null: false" on schema, or "required" on field configuration? That should take care of the "require a value to be selected" part if I'm not mistaken 🤔 Otherwise you could rely on count() in the validation permissions to require a specific amount of items to be selected

**Reply by JasonAGross** [reply]
Setting allow null to false and/or setting the field to be required has no impact. In both scenarios, either separately or combined, the collection can be created or updated without any options selected. 

I'm a bit lost on how I can use count() in the validation permissions. Is there somewhere else within the application that I need to go to establish validation rules before I can select them on the field itself? When looking at the validation screen my options are very limited. I can add a filter or a group and when I select to add a filter I can only choose between null or not null. 

![image](https://user-images.githubusercontent.com/1872357/180859564-6cdcee03-f630-4ebb-a19d-7ea6b336979b.png)

**Reply by rijkvanzanten** [reply]
What type of field is login options? JSON?

**Reply by JasonAGross** [reply]
> What type of field is login options? JSON?

Yes. It was created using the "Dropdown (Multiple)" option in the UI.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
